### PR TITLE
fix: restore docker push

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

**Comment by michielbdejong** [comment]
Restoring the docker-related steps that used to exist in https://github.com/interledgerjs/ilp-kit/blob/1b2995630cbe7b68ffaf4c9810fdf1938881f1e2/circle.yml

**Comment by michielbdejong** [comment]
It used to run docker build on every step, but I think that's a waste of circleci resources, so moved that to just the release branch

File: circle.yml [hunk 1]
```diff
@@ -23,6 +23,12 @@ deployment:
   production:
     branch: release
     commands:
+      # Build and push Docker image tagged latest and tagged with commit descriptor
+      - docker build -t interledgerjs/ilp-kit:latest .
+      - docker tag interledgerjs/ilp-kit:latest interledgerjs/ilp-kit:"$(git describe --tags)"
+      - sed "s/<AUTH>/${DOCKER_TOKEN}/" < "dockercfg-template" > ~/.dockercfg
```
Comment (emschwartz) [inline circle.yml]: If you just have the string `${DOCKER_TOKEN}` in the dockercfg-template, I think it'll grab the env var automatically

Comment (michielbdejong) [reply circle.yml]: We can address that in a follow-up issue, this is just to restore what was there before.

**Pull request merged**